runtime/sema/check_variable_declaration.go

```diff
@@ -280,6 +280,16 @@ func (checker *Checker) referencedVariables(expr ast.Expression) (variables []*V
 			variableRefExpr = rootVariableOfExpression(refExpr.Expression)
 		case *ast.IdentifierExpression:
 			variableRefExpr = &refExpr.Identifier
+		case *ast.IndexExpression:
+			// If it is a reference expression, then find the "root variable".
+			// As nested resources cannot be tracked, at least track the "root" if possible.
+			// For example, for an expression `a[b][c]`, the "root variable" is `a`.
+			variableRefExpr = rootVariableOfExpression(refExpr.TargetExpression)
+		case *ast.MemberExpression:
+			// If it is a reference expression, then find the "root variable".
+			// As nested resources cannot be tracked, at least track the "root" if possible.
+			// For example, for an expression `a.b.c`, the "root variable" is `a`.
+			variableRefExpr = rootVariableOfExpression(refExpr.Expression)
```

Comment:
This is an improvement to the static reference invalidation check.

Since now member-access/index-access can also produce a reference, they are also tracked now.

A nice side effect of this is, now attachment invalidation is also happening statically (see the new test case added in `runtime/tests/checker/reference_test.go` below)

Reply:
Great! 👌